base_layer/wallet/src/transaction_service/handle.rs

```diff
@@ -40,6 +40,10 @@ pub enum TransactionServiceRequest {
     GetPendingInboundTransactions,
     GetPendingOutboundTransactions,
     GetCompletedTransactions,
+    RegisterCallbackReceivedTransaction((unsafe extern "C" fn(*mut InboundTransaction))),
+    RegisterCallbackReceivedTransactionReply((unsafe extern "C" fn(*mut CompletedTransaction))),
+    RegisterCallbackMineCompletedTransaction((unsafe extern "C" fn(*mut CompletedTransaction))),
```

Comment:
I think we should call this callback `TransactionMined`. You have used the name we used for the test function that mines the transaction and I think the callback should be named based on the event that it represents.

Reply:
Updated in next commit